client/cli/src/main.rs

```diff
@@ -479,11 +480,19 @@ async fn main() {
         let cfg: UserClientConfig = load_from_file(&cfg_path).expect("Failed to parse config");
         let db = fedimint_rocksdb::RocksDb::open(db_path)
             .or_terminate(CliErrorKind::IOError, "could not open transaction db");
-        let db = Database::new(db, module_decode_stubs());
 
-        let rng = rand::rngs::OsRng;
+        let decoders = ModuleDecoderRegistry::new(cfg.clone().0.modules.into_iter().map(
+            |(id, module_cfg)| {
+                let module_gen = module_gens
+                    .get(module_cfg.kind())
+                    .expect("module kind not found in registry");
+                (id, module_gen.to_dyn_common().decoder())
```

Comment:
I think you could get away with just getting a reference to the common module gen instead of cloning, I should have documented it better:

https://github.com/fedimint/fedimint/blob/d20cb9750d8be9d38f9570a120af6d6a130fbf7c/fedimint-core/src/module/mod.rs#L466-L470

The `to_dyn_common` is useful when you can't deal with references, e.g. need to build a whole common registry.